wallet/src/lib.rs

```diff
@@ -337,33 +357,48 @@ mod wallet {
      * Cycle Management
      **************************************************************************************************/
     #[derive(CandidType)]
-    struct BalanceResult {
-        amount: u64,
+    struct BalanceResult<TCycles> {
+        amount: TCycles,
     }
 
     #[derive(CandidType, Deserialize)]
-    struct SendCyclesArgs {
+    struct SendCyclesArgs<TCycles> {
         canister: Principal,
-        amount: u64,
+        amount: TCycles,
     }
 
     /// Return the cycle balance of this canister.
     #[query(guard = "is_custodian_or_controller", name = "wallet_balance")]
-    fn balance() -> BalanceResult {
+    fn balance() -> BalanceResult<u64> {
         BalanceResult {
             amount: api::canister_balance(),
```

Comment:
Do you know what the behavior is of `api::canister_balance()` when the balance exceeds 64 bits?  Do you think this should call canister_balance128(), and exit with a similar error message ("call balance128") if so?

Reply:
It traps with a replica-defined error message. It probably should be defined on the wallet's end, just so the correct function name should be printed.